arcgis-ios-sdk-samples/Layers/Browse OGC API feature service/BrowseOGCAPIFeatureServiceViewController.swift

```diff
@@ -0,0 +1,234 @@
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = AGSMap(basemapStyle: .arcGISTopographic)
+            // Set the viewpoint to Daraa, Syria.
+            mapView.setViewpoint(AGSViewpoint(latitude: 32.62, longitude: 36.10, scale: 20_000))
+        }
+    }
+    /// The bar button to browse feature layers.
+    @IBOutlet var browseBarButtonItem: UIBarButtonItem!
+    
+    // MARK: Properties
+    
+    /// The most recent query job.
+    var lastQuery: AGSCancelable?
+    /// The Daraa, Syria OGC API feature service URL.
+    let defaultServiceURL = URL(string: "https://demo.ldproxy.net/daraa")!
+    /// The service metadata of feature collections from the loaded service.
+    var featureCollectionInfos = [AGSOGCFeatureCollectionInfo]()
+    /// The OGC feature collection info selected by the user.
+    var selectedInfo: AGSOGCFeatureCollectionInfo?
+    
+    /// The query parameters to populate features from the OGC API service.
+    let queryParameters: AGSQueryParameters = {
+        let queryParameters = AGSQueryParameters()
+        // Set a limit of 1000 on the number of returned features per request,
+        // because the default on some services could be as low as 10.
+        queryParameters.maxFeatures = 1_000
+        return queryParameters
+    }()
+    
+    // MARK: Methods
+    
+    /// Create and load the OGC API features service from a URL.
+    func loadService(url: URL) {
+        let service = AGSOGCFeatureService(url: url)
+        service.load { [weak self] error in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else if let infos = service.serviceInfo?.featureCollectionInfos {
+                self.featureCollectionInfos = infos
+                self.browseBarButtonItem.isEnabled = true
+            }
+        }
+    }
+    
+    /// Create a feature layer from the OGC feature collection table.
+    /// - Parameter info: The `AGSOGCFeatureCollectionInfo` selected by user.
+    func loadLayer(info: AGSOGCFeatureCollectionInfo) {
+        // Cancel if there is an existing query request.
+        lastQuery?.cancel()
+        // Remove existing layers.
+        mapView.map?.operationalLayers.removeAllObjects()
+        
+        let table = AGSOGCFeatureCollectionTable(featureCollectionInfo: info)
+        // Set the feature request mode to manual (only manual is currently
+        // supported). In this mode, you must manually populate the table -
+        // panning and zooming won't request features automatically.
+        table.featureRequestMode = .manualCache
+        
+        lastQuery = table.populateFromService(
+            with: queryParameters,
+            clearCache: false,
+            outfields: nil
+        ) { [weak self, table] _, error in
```

Comment:
Searching with `.cancel()` within the project, I don't see other samples nilify the `AGSJob` (or `AGSCancelable`) when they complete. However, there are a few samples set the job to nil at the top, together with the `.cancel()` call.

Do you want me to
- Set the job to nil at the top of the method
- Set the job to nil in the completion handler
- Don't make a change

I'm not sure the impact of not setting it to nil here. Can you explain a bit more?
Thanks!

Reply:
> I'm wondering if featureCollectionInfos should maybe be featureCollectionTables?

Not sure hmm. To me, holding an array of tables doesn't make a huge difference to holding infos. The design held the infos for the drop down list, so I followed.

Reply:
> I'm not sure the impact of not setting it to nil here.

It is simply a matter of being a good steward of the memory given to the app. At this point the cancellable is no longer needed, so it should be set to `nil` so that 1) It can be deallocated, and 2) so that `cancel()` isn't called on it the next time.

> there are a few samples set the job to nil at the top, together with the .cancel() call.

The important thing is to call `cancel()` on the current cancellable (if it exists) to stop the operation right away. Setting the associated property to `nil` is just extra work as the property's value is going to be updated with the cancellable for the new operation.

Does that make sense?

Reply:
Suggestion applied.